bigframes/dataframe.py

```diff
@@ -937,6 +912,177 @@ def obj_ref_rt_to_html(obj_ref_rt) -> str:
         html_string += f"[{row_count} rows x {column_count} columns in total]"
         return html_string
 
+    def _get_anywidget_bundle(self, include=None, exclude=None):
+        """
+        Helper method to create and return the anywidget mimebundle.
+        This function encapsulates the logic for anywidget display.
+        """
+        from bigframes import display
+
+        # Process blob columns if needed
+        self._cached()
+        df = self.copy()
+        if bigframes.options.display.blob_display:
+            blob_cols = [
+                series_name
+                for series_name, series in df.items()
+                if series.dtype == bigframes.dtypes.OBJ_REF_DTYPE
+            ]
+            for col in blob_cols:
+                df[col] = df[col].blob._get_runtime(mode="R", with_metadata=True)
+
+        # Create and display the widget
+        widget = display.TableWidget(df)
+        widget_repr_result = widget._repr_mimebundle_(include=include, exclude=exclude)
+
+        # Handle both tuple (data, metadata) and dict returns
+        if isinstance(widget_repr_result, tuple):
+            widget_repr = dict(widget_repr_result[0])  # Extract data dict from tuple
+        else:
+            widget_repr = dict(widget_repr_result)
+
+        # At this point, we have already executed the query as part of the
+        # widget construction. Let's use the information available to render
+        # the HTML and plain text versions.
+        widget_repr["text/html"] = widget.table_html
+
+        # Re-create the text representation from what we know.
+        opts = bigframes.options.display
+        with display_options.pandas_repr(opts):
+            import pandas.io.formats
+
+            # safe to mutate this, this dict is owned by this code, and does not affect global config
+            to_string_kwargs = (
+                pandas.io.formats.format.get_dataframe_repr_params()  # type: ignore
+            )
+            if not self._has_index:
+                to_string_kwargs.update({"index": False})
+            repr_string = widget._cached_data.to_string(**to_string_kwargs)
+
+        lines = repr_string.split("\n")
+        row_count = widget.row_count
+        if row_count is not None and row_count > len(widget._cached_data):
+            lines.append("...")
+
+        lines.append("")
+        column_count = len(self.columns)
+        lines.append(f"[{row_count or '?'} rows x {column_count} columns]")
+        widget_repr["text/plain"] = "\n".join(lines)
```

Comment:
Likewise, this looks duplicated as well. Please get the common parameters out and refactor to a shared function.